src/MassTransit/DependencyInjection/Configuration/ScopedFilterSpecificationObserver.cs

```diff
@@ -12,12 +12,17 @@ public class ScopedFilterSpecificationObserver :
         IPublishPipeSpecificationObserver
     {
         readonly Type _filterType;
+        readonly CompositeFilter<Type> _messageTypeFilter;
         readonly IServiceProvider _provider;
 
-        public ScopedFilterSpecificationObserver(Type filterType, IServiceProvider provider)
+        public ScopedFilterSpecificationObserver(Type filterType, IServiceProvider provider, CompositeFilter<Type> messageTypeFilter)
         {
             _filterType = filterType;
             _provider = provider;
+            _messageTypeFilter = messageTypeFilter;
+            _messageTypeFilter.Excludes += type => type.HasInterface<Fault>();
```

Comment:
Oh, hmm. I see what you've done here. Do we have tests that ensure the scoped filters aren't called for faulted messages?

Reply:
https://github.com/MassTransit/MassTransit/blob/6ec989a6aec2e2abbe891379aed44e30ea68697c/tests/MassTransit.Containers.Tests/Common_Tests/Common_ScopePublish.cs#L335

Reply:
Probably should add `type.HasInterface<ReceiveFault>();` as well, since that's another fault type which can be published.

Reply:
Mentally thinking about this, I'm fairly sure this was only meant to apply to published faults but somehow is being applied to all message types. Hmm.

Reply:
is it for consumer or send?